### PR TITLE
Settings Page

### PR DESCRIPTION
**Pull request by Daniel3886**
Basic structure of Settings Page.

### CODE REVIEW

File: client/src/components/layouts/HomeLayout.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ const HomeLayout: React.FC<Props> = ({
   centeredSx,
   ...rest
 }) => {
-  const contentWidth = 1000;
+  const contentWidth = 100;
```
Comment (Softjey) [inline client/src/components/layouts/HomeLayout.tsx:20]: don'y change home layout

File: client/src/pages/ChangePinCode.tsx
Comment (Softjey) [inline client/src/pages/ChangePinCode.tsx:1]: all components have to be in components folder

File: client/src/pages/ChangeTheme.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  padding: 7,
+  '& .MuiSwitch-switchBase': {
+    margin: 1,
+    padding: 0,
+    transform: 'translateX(6px)',
+    '&.Mui-checked': {
+      color: 'yellow',
+      transform: 'translateX(22px)',
+      '& .MuiSwitch-thumb:before': {
+        backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
```
Comment (Softjey) [inline client/src/pages/ChangeTheme.tsx:19]: don't use text hardcoded icons, use mui icons components instead 
https://mui.com/material-ui/material-icons/

File: client/src/pages/ChangeTheme.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    '&::before': {
+      content: "''",
+      position: 'absolute',
+      width: '100%',
+      height: '100%',
+      left: 0,
+      top: 0,
+      backgroundRepeat: 'no-repeat',
+      backgroundPosition: 'center',
+      backgroundImage: `url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="20" width="20" viewBox="0 0 20 20"><path fill="${encodeURIComponent(
```
Comment (Softjey) [inline client/src/pages/ChangeTheme.tsx:42]: same here
https://mui.com/material-ui/material-icons/

File: client/src/pages/SettingsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import ChangeTheme from './ChangeTheme';
+import ChangeUserInfo from './ChangeUserInfo';
+import DeleteAccount from './DeleteAccount';
+
+const SettingsPage: React.FC = () => {
+  return (
+    <HomeLayout>
+      <Box sx={{ bgcolor: 'background.paper', color: 'text.primary', height: '100vh', p: 1 }}>
+        <List sx={{ width: '100%', maxWidth: 350, paddingTop: 0, bgcolor: 'background.paper' }}>
+          <Box sx={{ mb: 1 }}>
```
Comment (Softjey) [inline client/src/pages/SettingsPage.tsx:14]: use flexbox instead of boxes with margin

File: client/src/types/api/entities/user.ts
Comment (Softjey) [inline client/src/types/api/entities/user.ts:1]: don't chnage this file

File: server/package-lock.json
Comment (Softjey) [inline server/package-lock.json:1]: don't chnage this file

**Comment by Daniel3886** [comment]
Hello Softjey! I have reviewed your email and I have changed everything as
you requested. I couldn't quite understand what you meant by not changing a
file called "server/package-lock.json
<https://github.com/Softjey/react_Examix/pull/47#discussion_r1623643557>".
The issue is that I never touched the file and changed anything there
If possible please provide more details about the issue
Thank you for understanding.
Daniel Czerczon

вс, 2 июн. 2024 г. в 21:59, Softjey ***@***.***>:

**Review by Softjey — APPROVED** [review]

**Pull request merged**